src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanJobService.java

```diff
@@ -60,87 +67,54 @@ public ActionPlanJob findActionPlanJob(final UUID actionPlanJobId) {
     return actionPlanJobRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
   }
 
-  public List<ActionPlanJob> createAndExecuteAllActionPlanJobs() {
-    final List<ActionPlanJob> executedJobs = new ArrayList<>();
-    actionPlanRepo
-        .findAll()
-        .forEach(
-            actionPlan -> {
-              final Date lastExecutionTime =
-                  new Date(
-                      nowUTC().getTime() - appConfig.getPlanExecution().getDelayMilliSeconds());
-              if (actionPlan.getLastRunDateTime() == null
-                  || actionPlan.getLastRunDateTime().before(lastExecutionTime)) {
-                ActionPlanJob job =
-                    ActionPlanJob.builder()
-                        .actionPlanFK(actionPlan.getActionPlanPK())
-                        .createdBy(CREATED_BY_SYSTEM)
-                        .build();
-                job = createAndExecuteActionPlanJob(job);
-                if (job != null) {
-                  executedJobs.add(job);
-                }
-              } else {
-                log.with("action_plan_id", actionPlan.getId())
-                    .with("action_plan_pk", actionPlan.getActionPlanPK())
-                    .debug("Job for plan has been run since last wake up - skipping");
-              }
-            });
-    return executedJobs;
-  }
-
   /**
-   * Create a new ActionPlanJob record and execute createActions stored procedure.
-   *
-   * @param actionPlanJobTemplate template {@link ActionPlanJob} to create and execute.
-   * @return ActionPlanJob that was created or null if it has not been created.
+   * Create a new ActionPlanJob and create associated actions for all action plans with cases in the
+   * action.case table
    */
-  public ActionPlanJob createAndExecuteActionPlanJob(final ActionPlanJob actionPlanJobTemplate) {
-    final Integer actionPlanPK = actionPlanJobTemplate.getActionPlanFK();
-    final ActionPlan actionPlan = actionPlanRepo.findOne(actionPlanPK);
-
-    if (actionPlan == null) {
-      log.with("action_plan_pk", actionPlanPK).debug("Action plan is null");
-      return null;
-    }
-
-    if (actionCaseRepo.countByActionPlanFK(actionPlanPK) == 0) {
-      log.with("action_plan_pk", actionPlanPK).debug("No open cases for action plan");
+  @Transactional
+  public void createAndExecuteAllActionPlanJobs() {
+    List<ActionPlan> actionPlans = actionPlanRepo.findAll();
+    actionPlans.forEach(
+        actionPlan -> {
+          if (hasActionableCases(actionPlan)) {
+            createAndExecuteActionPlanJob(actionPlan);
+          }
+        });
+  }
 
-      return null;
-    }
+  private boolean hasActionableCases(ActionPlan actionPlan) {
+    return actionCaseRepo.countByActionPlanFK(actionPlan.getActionPlanPK()) > 0;
```

Comment:
if this is the only place this repo method is used, consider refactoring it to be `exists` instead of `count`